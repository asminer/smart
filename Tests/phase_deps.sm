


ph int zero   := 0;
ph int one    := 1;
ph int two    := 2;

ph int zpo    := zero + one;
ph int zpopt  := zero + one + two;

print_deps(zero);
print_deps(one);
print_deps(two);

print_deps(zpo);
print_deps(zpopt);

ph int zoo    := choose(zero:1, one:2);
ph int zooot  := choose(zero:1, one:2, two:1);

ph int mzo    := max(zero, one);

print_deps(zoo);
print_deps(zooot);
print_deps(mzo);

ph int zpinf  := zero + infinity;
ph int opinf  := one + infinity;

print_deps(zpinf);
print_deps(opinf);

ph int tricky1 := zpinf + opinf;

print_deps(tricky1);

print("These should print warnings:\n");

ph int warn1 := zpo + zpopt;
ph int warn2 := zoo + zooot;
ph int warn3 := zpo + zooot;
ph int warn4 := choose(zpo:1, zooot:2);
ph int warn5 := max(zpo, zero, one);
ph int warn6 := min(zoo, zooot);
