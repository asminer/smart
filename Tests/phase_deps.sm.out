Phase type distribution zero has 1 variable dependencies:
	zero
Phase type distribution one has 1 variable dependencies:
	one
Phase type distribution two has 1 variable dependencies:
	two
Phase type distribution zpo has 3 variable dependencies:
	zpo
	zero
	one
Phase type distribution zpopt has 4 variable dependencies:
	zpopt
	zero
	one
	two
Phase type distribution zoo has 3 variable dependencies:
	zoo
	zero
	one
Phase type distribution zooot has 4 variable dependencies:
	zooot
	zero
	one
	two
Phase type distribution mzo has 3 variable dependencies:
	mzo
	zero
	one
Phase type distribution zpinf has 3 variable dependencies:
	zpinf
	zero
	infinity (ignored)
Phase type distribution opinf has 3 variable dependencies:
	opinf
	one
	infinity (ignored)
Phase type distribution tricky1 has 7 variable dependencies:
	tricky1
	zpinf
	zero
	infinity (ignored)
	opinf
	one
	infinity (ignored)
These should print warnings:
WARNING in file phase_deps.sm near line 39:
	operands in phase-type addition are not independent;
	phase-type model will incorrectly treat them as such.
	Common dependencies:
		zero
		one
WARNING in file phase_deps.sm near line 40:
	operands in phase-type addition are not independent;
	phase-type model will incorrectly treat them as such.
	Common dependencies:
		zero
		one
WARNING in file phase_deps.sm near line 41:
	operands in phase-type addition are not independent;
	phase-type model will incorrectly treat them as such.
	Common dependencies:
		zero
		one
WARNING:
	Common dependencies in parameters to choose():
		zero
		one
WARNING in file phase_deps.sm near line 42:
	Function choose requires independent parameters.
	Phase-type model will be incorrect.
WARNING:
	Common dependencies in parameters to max():
		zero
		one
WARNING in file phase_deps.sm near line 43:
	Function max requires independent parameters.
	Phase-type model will be incorrect.
WARNING:
	Common dependencies in parameters to min():
		zero
		one
WARNING in file phase_deps.sm near line 44:
	Function min requires independent parameters.
	Phase-type model will be incorrect.
